tests/NCronJob.Tests/JobOptionBuilderTests.cs

```diff
@@ -59,4 +59,43 @@ public void ShouldAddMultipleCronJobsEvenWithoutParameters()
         options[1].EnableSecondPrecision.ShouldBeFalse();
         options[1].Parameter.ShouldBeNull();
     }
+
+    [Fact]
+    public void ShouldCreateMultipleJobsWithoutAnd()
+    {
+        var builder = new JobOptionBuilder();
+        builder.WithCronExpression("* * * * *")
+            .WithParameter("foo");
+
+        builder.WithCronExpression("0 * * * *")
+            .WithParameter("bar");
+
+        var options = builder.GetJobOptions();
+
+        options.Count.ShouldBe(2);
+        options[0].CronExpression.ShouldBe("* * * * *");
+        options[0].EnableSecondPrecision.ShouldBeFalse();
+        options[0].Parameter.ShouldBe("foo");
+        options[1].CronExpression.ShouldBe("0 * * * *");
+        options[1].EnableSecondPrecision.ShouldBeFalse();
+        options[1].Parameter.ShouldBe("bar");
+    }
+
+    [Fact]
+    public async Task ShouldHandleConcurrentAccessesGracefully()
```

Comment:
Is this a realistic case?
Normally there is a single thread configuring this once.

Reply:
I added the test to verify that the Tasks were all being created and I discovered that 20% of the time we were not getting 100 tasks. It made it obvious that it wasn't thread safe. That's why I changed the `JobOptionBuilder.jobOptions` to a ConcurrentQueue<JobOption>. 
>Is this a realistic case?
Normally there is a single thread configuring this once.

Yes currently only one would be created at a time so no, it's not a use case... but I was thinking ahead with features like auto Job registrations ;-) 
Probably a case of premature optimization. It doesn't hurt to keep the test unless we also revert the jobOptions to a List.